ImageRater/MainWindow.xaml.cs

```diff
@@ -14,16 +13,30 @@ public partial class MainWindow : Window
     public MainWindow()
     {
         using (var dialog = new FolderBrowserDialog())
-
-            while (!Directory.Exists(dialog.SelectedPath))
+        {
+            dialog.Description = "Please select the folder that contains your input images.";
```

Comment:
Looks good tho :).
You can also directly declare the parameters when creating the instance:
```
        using (var dialog = new FolderBrowserDialog()
        {
            Description = "Please select the folder that contains your input images.",
            UseDescriptionForTitle = true,
        })
```
This is just my personal taste, it makes the declaration more readable imho.